src/Compilers/CSharp/Portable/Utilities/IValueSetFactory.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// A value set factory, which can be used to create a value set instance.  A given instance of <see cref="IValueSetFactory"/>
+    /// supports only one type for the value sets it can produce.
+    /// </summary>
+    internal interface IValueSetFactory
+    {
+        /// <summary>
+        /// Returns a value set that includes all values of its type.
+        /// </summary>
+        IValueSet All { get; }
```

Comment:
>All [](start = 18, length = 3)

Are `All` or `None` used, from here or `IValueSetFactory<T>`?

Reply:
Principally in tests.  I'll remove them.

---
In reply to: [378550229](https://github.com/dotnet/roslyn/pull/41503#discussion_r378550229) [](ancestors = 378550229)